awscrt/http.py

```diff
@@ -144,6 +152,73 @@ def on_incoming_headers_received(headers, response_code, has_body):
 
         return request
 
+
+'''
+TODO: Server Connection class,and configure the server connection.
+'''
+
+
+class HttpServer(object):
+    """
+    Represents an Http server. Everything in this class is non-blocking.
+    """
+    __slots__ = ('_bootstrap', '_tls_connection_options', '_on_incoming_connection',
+                 '_on_destroy_complete', '_native_handle')
+
+    # don't call me, I'm private
+    def __init__(self, bootstrap, on_incoming_connection, on_destroy_complete, tls_connection_options):
+        assert isinstance(bootstrap, ServerBootstrap)
+        assert tls_connection_options is None or isinstance(
+            tls_connection_options, TlsConnectionOptions)
+
+        for slot in self.__slots__:
+            setattr(self, slot, None)
+
+        self._bootstrap = bootstrap
+        self._tls_connection_options = tls_connection_options
+        self._on_incoming_connection = on_incoming_connection
+        self._on_destroy_complete = on_destroy_complete
+        self._native_handle = None
+
+    @staticmethod
+    def new_server(bootstrap, host_name, port, socket_options, on_incoming_connection,
+                   on_destroy_complete=None, tls_connection_options=None):
+        """
+        Create a new server listener, binding to the host_name and port. 
+        When a new connection is received, the on_incoming_connection cb will be fired, a new ServerConnection obj will be created.
+        The aws_py_http_connection_configure_server need to be called from the callback to configure the ServerConnection 
+        """
+        assert tls_connection_options is None or isinstance(
+            tls_connection_options, TlsConnectionOptions)
+        assert host_name is not None
+        assert port is not None
+        assert socket_options is not None and isinstance(
+            socket_options, SocketOptions)
+        assert on_incoming_connection is not None
+        if tls_connection_options is not None:
+            internal_conn_options_handle = tls_connection_options._internal_tls_conn_options
+        else:
+            internal_conn_options_handle = None
+        server = HttpServer(bootstrap, on_incoming_connection,
+                            on_destroy_complete, tls_connection_options)
+        server._native_handle = _aws_crt_python.aws_py_http_server_create(
+            bootstrap._internal_bootstrap, on_incoming_connection, on_destroy_complete, host_name, port, socket_options, internal_conn_options_handle)
+        return server
+
+    def release(self):
+        """
+        release the server, no more connections will be accepted, when the server finishes destroy process, the on_destroy_complete will be invoked
```

Comment:
maybe rename this `close()`
In C, release() is a combination "shut this thing down and release my refcount" call.
In python, users don't think about refcounts so lets use simple terms

Also, the docstring should mention the fact that a future is being returned. Also, we might want to rename remove the `on_destroy_complete` callback that's passed into the constructor, since it doesn't always fire, and we provide the user this future if they want to learn about "shutdown complete"